### DISCUSSION TITLE
For Toolmakers — let us know here!

### DISCUSSION THREAD

**Discussion by Myndex** [Myndex/SAPC-APCA#51] [Show and tell]
This thread is only to list links to current tools that use APCA, APCA-W3, Bridge-PCA, SAPC, or SACAM technologies. 

If you have a tool you'd like us to checkout and list here, please feel free to post the link and a bit of information about it in this thread, or message me direct. Thank you!!!

### The contents of this post are periodically moved to the [Master Public List](https://git.apcacontrast.com/documentation/thirdpartytools)

-----
# _Third Party APCA Tools_
Please post a message to have your tool added to the list!

Take note that some of these tools are works in progress, and not all of them have been fully analyzed for accuracy or conformance to APCA standards. We will be starting qualification testing in the near future.

### [Tailwind](https://uicolors.app/create)
- It's a very cool tool, with a plugin for Figma, make sure to use the latest version as it is now updated to use the current APCA constants. This tool can show either WCAG2 or APCA.

### [Stark APCA Beta](https://www.getstark.co/blog/stark-APCA-beta)
- Yes, Stark has added APCA to their toolset as an option! It's available across the product line, and included in the free versions. Make sure to use the latest version which is updated to use the current APCA constants. This tool can show either WCAG2 or APCA.

### [Accessible Palette](https://AccessiblePalette.com)
- A very nice color selector by Eugene, see [his article as well](https://wildbit.com/blog/accessible-palette-stop-using-hsl-for-color-systems) This shows both WCAG 2 and APCA.

### [Huetone](https://huetone.ardov.me/?palette=N4IgdghgtgpiBcICiAPaAHANjABABQmwBci4AaEACwFcYBnBAbVElgRAHEAnGATxAoBjAPaZhXBvEYgAxADMAbIsUDZMAIwaNqmQCMAJgYM6I606Z0AOAEw2bOhdceOdAFgDsHjzusBOP3466pbBwSAAugC%2BZCzQcIh4AJZgANaqImISTLJyudZy7jq5%2Bu4wAMxFuhCuggAMRcoQFRTy1tYQlgCsOrq%2BmgquVnVyrr46nZ217u06ZZNlYVEx4HHsAEow%2Bumi4pLS8rmuMOQ5cvq%2Bgify%2BhDuvmMtue7uCluP6mX6ruo90%2BX1LUsfRg1gBsk6ZVqcnUYJkcymtXqS1ibEQAE0YJgxAB3baZPanOSdGCWBqGXSkx5lCAFZpqSyWJoKHSCXQKdy1H4tXy6Vz6Tk6dwfSy1QYtVxtRGw9rqGFI6Io%2BKcHgwMB43bZeRQ3wwB6yQRyXW6XQ6XwQVnuSmyBQ8splK4S3yuKE%2BAqIwotdRE0Gw9SuSGIoKc3SB5ErVEgAAqMEI6qyUjU%2BRpch6ChjcitMhegl8%2BiutU6IcEb1ktQUljkEGZLSmCkEpR0iNcRJLMk51nKHtLst0oIiCvDSoAQphaHGCfJm0oUy1DPpcjPZBAIFVclY-Lo1%2BKpgbM%2BpBJ19II918-GLS-4FMTGwp1AoyiaouEKERhGB6NlJqo5aofaoA6ozpgpMYIKIGFAcmCIrykAA)
- Cool interface under development, lots of options. This tool can show either WCAG2 or APCA.

### [Polypane Color Contrast](https://colorcontrast.app/#fg=%23107db5&bg=%23fff&level=aa&format=rgb&algo=APCA&filter=none)
- Killian's Contrast app, can show either WCAG2 or APCA.
    - _(Killian was the key designer behind the updated APCA demo tool at Myndex.)_

### [Cliam's Contrast](https://cliambrown.com/contrast/)
- A neat color checker that allows you to click to find a given contrast level.

### [Ambient Color Generation Tool](https://ambient-rvx.web.app)
- This tool is designed to generate contrast ready color pallets built specifically for UI/UX design. Unlike other palette generation tools, Ambient generates the colors based on the primary color.

### [Atmos](https://atmos.style/contrast-checker)
- a contrast checker that is part of the Atmos design system.

### [contrast-suggester.com](https://contrast-suggester.com) 
Also a [GitHub repo too](https://github.com/Munawwar/apca-color-contrast)
- Suggests a contrasting color for a given color, font size, and font weight. 

### [Zebra for Figma](https://www.figma.com/community/plugin/806578669827234193)
- Dan Hollick's Zebra contrast plug in for Figma.

### [Caravage for Figma](https://www.figma.com/community/plugin/1059851135811973410/Caravage)
- A Figma plugin to check the contrast accessibility of the selected text. It is still in beta, the developer @simonbrtrnd  mentions any feedback is appreciated.

### [RandomA11y APCA](https://randoma11y.com/apca?color=%23c91313)
- this is an interesting take on a tool that gives you the opposite color for the color that you choose.

### [contrast-checker Vercel app](https://contrast-checker.vercel.app/000000e6/d40511/ffcc00/ffcc004d/ffffff/0000000d/0000001a/76BD22/007C39)
- This WCAG 2 checker also shows APCA so you can see the relative differences.

### [Contrast Tools](https://contrast.tools)
- Nice clean interface, easy to use. This tool can show either WCAG2 or APCA.

### [Contrast Checker](https://contrast-checker.bellette.com.au/)
- A clever interface that demonstrates the key concepts as well.

### [APCA Flip Black/White](https://apca.digitallymarked.com)
Also, [the GitHib repo](https://github.com/digitallymarked/apca-calculator)

- This is a simple but informative interface, text is only black or white, and it's using APCA to determine when to flip between the two.

### [Japanese Experimental](https://develop.apca-calculator-ja.pages.dev)
Also, [the GitHib repo](https://github.com/shunito/apca-calculator-ja)

- This is a Japanese experimental version, it's part of the project for working on aligning Japanese writing systems.

----
### [color-contrast.dev](https://www.color-contrast.dev)
- this is the first third-party implementation I've seen of Bridge PCA, the drop in replacement that's backwards compatible with WCAG2. Nice simple input interface that also gives you suggested colors to improve contrast.

### [Color Contrast iOS app](https://apps.apple.com/app/color-contrast/id1095478187?platform=iphone)
- This an iOS app to measure the contrast between two colours in a screenshot or mobile website. The APCA algorithm is available as an option, with the Bridge PCA being used to show the modified WCAG ratio.

-----
### [Chrome Dev Tools](https://developer.chrome.com/blog/new-in-devtools-89/#apca)
- Available as an experimental feature, note that some versions maybe using obsolete constants and therefore may give incorrect results.

### [Microsoft Edge Dev Tools](https://docs.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/whats-new/2021/01/devtools#new-color-contrast-calculation---advanced-perceptual-contrast-algorithm-apca)
- Available as an experimental feature, note that some versions maybe using obsolete constants and therefore may give incorrect results.

-----
### [Color.js](https://colorjs.io)
- Comprehensive JS color library by Lea Verou and Chris Lilly.
- See the [contrast demo page](https://colorjs.io/apps/blackwhite/)

### [SASS-APCA](https://github.com/gfellerph/sass-apca)
- a sass library

### [A11y Color Contrast](https://deno.land/x/a11y_color_contrast)
- a typescript library

### [OxyProps/BrickProps](https://docs.oxyprops.com/en/features/global-colors/)
- this is the design system plug-in for WordPress, I'm not quite sure how they're integrating a PCA but they do mention in the docs.

## _Works in Progress_
These are some works in progress, functionality was not complete at the last time I checked, and or may have math or implementation errors. If your tool is on this portion of the list, please feel free to reach out, I'm more than happy to assist.

### [Leonardo](https://leonardocolor.io/#)
- The adaptive color theme function has an APCA option, however APCA has not yet been integrated into the color contrast tool or the dataviz tool. The Leonardo tool is pretty cool as it stands.

### [ROKING-A11Y APCA Evaluator](https://hrobertking.github.io/roking-a11y/apca.htm)
- Has a visible look up table which changes based on the contrast. Has a pending pull request due to some errors in terms of the exponents and look up.

### [AdamSzakal APCA-contrast](https://acpa-contrast.netlify.app)
- The repo is spelled wrong (oops!): https://github.com/AdamSzakal/ACPA-contrast
Basically looks like a work in progress, does not show polarity in the results.

### [Visual Contrast Figma Plugin](https://www.figma.com/community/plugin/1090308131937420683)
- Does not seem to work correctly??

### https://jolly-feynman-28ea37.netlify.app
- https://github.com/tomreinert/contrast-calculator-2

-----
### [APCA W3 Tool](https://www.myndex.com/APCA/)
- The main canonical demonstration tool. Public beta. The is the candidate for the future WCAG 3 contrast method.
    - Note that this is not fully backwards compatible with WCAG 2 contrast math due to the non-perceptual nature of WCAG 2.
    - Legacy version available on [a backup link](https://myndex.github.io/SAPC-APCA/WEBTOOLS/APCA/) at this repo.
    - The dedicated domain name is [apcacontrast.com](https://apcacontrast.com)

### [Color Vision Deficiency (ColorBlind) Simulator](https://www.myndex.com/CVD/)
- The CVD sim, a clinically accurate simulation of various forms of color insensitive vision (aka color blind), most of the sims are derived from or based directly on the clinically accurate Brettel model. Allows you to process a screenshot of your page design and view several CVD types at once..

### [Bridge-PCA](https://www.myndex.com/BPCA/)
- Bridge PCA is a modified tool that is backwards compatible with WCAG 2 contrast, but using APCA technology.
    - In order to be backwards compatible, some design flexibility is given up, but accessibility is substantially improved over WCAG 2.
    -  The dedicated domain name is [bridgepca.com](https://bridgepca.com)

### [SAPC Development Site](https://www.myndex.com/SAPC/)
- The SAPC research demonstration tool, more detailed results than the APCA tool above, and a clickable "research mode" with technology demonstrators, interactive experiments, and much data about the colors in use.
- NOTE: This tool may not always match the main tool depending on current experimental protocols.

-----

### [APCA W3](https://github.com/Myndex/apca-w3)
And the npm install is   `  npm i apca-w3  `

### [Bridge-PCA](https://github.com/Myndex/bridge-pca)
And the npm install is   `  npm i bridge-pca  `

### [The Main SAPC APCA SARCAM](https://github.com/Myndex/SAPC-APCA)
This repo right here...

### [Guidelines for Accessible Readable Design](https://github.com/A11yReadTech/GARD)
In development at Accessible Reading Technologies: GARD is a developing set of readability design guidelines.

### [deltaPhiStar](https://github.com/Myndex/deltaphistar)
An alternate and highly simplified contrast algorithm. Still fairly perceptually uniform though less accurate for very high and very low contrast, and it's not polarity aware.

### [See Stars](https://github.com/Myndex/seestars)
Myndex microlibrary for CIE Lstar, for use with deltaPhiStar

### [Fancy Font Flipping](https://github.com/Myndex/fancyfontflipping)
Related to APCA methods, but separate—this is a demonstrator for when to flip a text color from black to white, based on the luminance of the background. CodePen also available, and also read [this Gist article.](https://gist.github.com/Myndex/e1025706436736166561d339fd667493#lets-flip-for-color)

### [Color Parsley](https://github.com/Myndex/colorparsley)
A lightweight but versatile color parsing utility with no dependencies
_NPM Install:_   ` npm i colorparsley `

----
### https://linktr.ee/Myndex
This is a linktree of articles and information regarding APCA.

### [Why APCA](https://git.apcacontrast.com/documentation/WhyAPCA)™ 
A brief overview of APCA. This is a good place to start if you are new to perceptually uniform contrast.

### [Myndex Color & Contrast Resources](https://git.myndex.com/)
A catalog of contrast and color resources.

----
## Accessibility Statement for Beta Testers
**Beta testers may wish to include the following statement in their sites boilerplate:**

There are additional versions in the [Accessibility Statements file](https://github.com/Myndex/SAPC-APCA/blob/master/documentation/accessibilitystatement.md)

-----
### This website is using the [APCA](https://git.apcacontrast.com/documentation/WhyAPCA)™ readability guidelines for determining visual contrast of text and non-text elements.

This site's visually-readable primary content is designed to meet or exceed the conformance level of: <insert bronze/silver/gold> as defined in version: <insert APCAversion> of the beta [APCA Readability Criteria (ARC)](https://readtech.org/ARC/), which provide improved visual readability, enhancing the actual accessibility for users of this website. If you have questions, concerns, or problems, please contact the accessibility manager at <insert Accessibility manager's contact info here>

-----

<sub>Copyright © 2021-2023 by Andrew Somers. All Rights Reserved. Patent(s) Pending.</sub>

**Comment by Munawwar** [comment]
[contrast-suggester.com](https://contrast-suggester.com) - Suggests a contrast color for a given color, font size and font weight.

It's uses HSLuv color space for finding contrast shades, unlike any in this list so far.

[Git repo link](https://github.com/Munawwar/apca-color-contrast)

**Comment by simonbrtrnd** [comment]
[Caravage](https://www.figma.com/community/plugin/1059851135811973410/Caravage) a Figma plugin to check the contrast accessibility of the selected text. It is still in beta, any feedback is appreciate 🙏

**Reply by simonbrtrnd** [reply]
Previous link is dead 💀
New link →  https://www.figma.com/community/plugin/1413184718185819161/caravage

**Comment by cgrobb** [comment]
https://github.blog/2022-06-14-accelerating-github-theme-creation-with-color-tooling/ does NOT include APCA, but should. Looks like contrast is via [color2k](https://github.com/ricokahler/color2k).

**Comment by accessresource** [comment]
[Color Contrast](https://apps.apple.com/app/color-contrast/id1095478187?platform=iphone) is an iOS app to measure the contrast between two colours in a screenshot or mobile website. The APCA algorithm is available as an option, with the Bridge PCA being used to show the modified WCAG ratio. Any feedback is appreciated.

**Comment by Myndex** [comment]
I came across this interesting tool: https://randoma11y.com/apca?color=%23c91313

**Reply by Myndex** [reply]
And another:  https://www.color-contrast.dev

**Comment by gfellerph** [comment]
A sass implementation of the APCA: https://github.com/gfellerph/sass-apca. Feedback very welcome.

**Comment by VojtechVidra** [comment]
We have implemented WCAG 2 as well as APCA contrast checker in [Atmos](https://atmos.style) - Everything you need to create color palettes.

Can be seen live at https://atmos.style/contrast-checker

**Comment by brainshift-design** [comment]
Hi, my Figma plugin HCL color has been supporting APCA for the past few years! 🤓 https://www.figma.com/community/plugin/889859275438287504

My other Figma plugin, Generator, also has a node that can check both WCAG2 and APCA color contrast: https://www.figma.com/community/plugin/899028246731755335

**Comment by drwpow** [comment]
👋 Would love to request permission to use of APCA in [Cobalt](https://cobalt-ui.pages.dev/), open-source (MIT) tooling for the [W3C Design Token format](https://design-tokens.github.io/community-group/format/). Specifically, I’m thinking about [adding “tests” to the tool](https://github.com/drwpow/cobalt-ui/issues/214), that run automatic contrast checkers on your design tokens so you can catch regressions. Being able to use APCA would be very handy for designers managing design systems who want to go the extra mile for contrast compliance (and since they can specify their fonts in design tokens as well, they could even test those too).

Love your work, and love this tool!

**Comment by cyrez** [comment]
Hello @Myndex !

I think i've forgotten to mention my Sass library for auto-generate color contrast with APCA algo here...

There it is: 
**A11y APCA Sass Library**
https://github.com/cyrezdev/a11y-apca-sass

**Comment by tbayramov** [comment]
Hello @Myndex ,

My wife @tatyanabayramova and I made an accessible color contrast checker which uses both APCA and WCAG 2. It has Live Preview, minimum font size table (based on the APCA value), and dynamic level descriptions. It also lets users know if a color combination fails the APCA evaluation, even when they check in WCAG mode (to raise awareness).

Curious to hear your thoughts and hope to be included in the tool list :)

https://a-plus.tech/tools/contrast-checker/

Warmest wishes.
